### PR TITLE
ci: update terraform scripts for oracle/rhel/rockylinux 9.4

### PR DESCRIPTION
**Pull request by yangchiu**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/9217

#### What this PR does / why we need it:

update terraform scripts for oracle/rhel/rockylinux 9.4

For `Oracle` on `amd64`, using `ProComputers Oracle-Linux-9.4-x86_64-Minimal-8GiB-HVM-20240525_061002-b26dc6a6-ce9a-4622-804d-285676f4`
For `Oracle` on `arm64`, no image is available on AWS

For `RHEL` on `amd64`, using `RHEL-9.4.0_HVM-20240718-x86_64-2-Hourly2-GP3`
For `RHEL` on `amr64`, no suitable image was found. Both `RHEL-9.4.0_HVM-20240718-arm64-2-Hourly2-GP3` and `(SupportedImages) - RHEL 9 ARM64 LATEST - 20240610-69b70f26-9f08-432b-a47e-9d582f69e916` failed to setup the cluster. More investigation needed.

For `Rockylinux` on `amd64`, using `Rocky-9-EC2-Base-9.4-20240523.0.x86_64-3f230a17-9877-4b16-aa5e-b1ff34ab206b`
For `Rockylinux` on `arm64`, no suitable image was found. `Rocky-9-EC2-Base-9.4-20240509.0.aarch64` failed to setup the cluster. More investigation needed.

### CODE REVIEW

File: test_framework/terraform/aws/oracle/user-data-scripts/provision_k3s_agent.sh.tpl [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
 
-sudo systemctl stop firewalld
-sudo systemctl disable firewalld
-
 sudo yum update -y
 sudo yum group install -y "Development Tools"
 sudo yum install -y iscsi-initiator-utils nfs-utils nfs4-acl-tools cryptsetup
 sudo systemctl -q enable iscsid
 sudo systemctl start iscsid
+sudo systemctl disable nm-cloud-setup.service nm-cloud-setup.timer
```
Comment (derekbit) [inline test_framework/terraform/aws/oracle/user-data-scripts/provision_k3s_agent.sh.tpl:8]: What's the purpose of disabling nm-cloud-setup?

Comment (yangchiu) [reply test_framework/terraform/aws/oracle/user-data-scripts/provision_k3s_agent.sh.tpl:8]: Otherwise k3s-agent service won’t start.

Comment (derekbit) [reply test_framework/terraform/aws/oracle/user-data-scripts/provision_k3s_agent.sh.tpl:8]: Can you leave a comment in the above of the change ? Thank you.

Comment (yangchiu) [reply test_framework/terraform/aws/oracle/user-data-scripts/provision_k3s_agent.sh.tpl:9]: Comment added. Thank you!

**Review by khushboo-rancher — APPROVED** [review]

**Comment by derekbit** [comment]
We only have amd64 images for the three OS `Oracle`, `RHEL` and `Rockylinux`.
Should we rollback to 9.3 instead?
We can update them to 9.4 while releasing v1.7.1.
WDYT?

cc @yangchiu @khushboo-rancher

**Comment by derekbit** [comment]
We only have amd64 images for the three OS `Oracle`, `RHEL` and `Rockylinux`.
Should we rollback to 9.3 instead?
We can update them to 9.4 while releasing v1.7.1.
WDYT?

cc @innobead  @khushboo-rancher

**Comment by innobead** [comment]
> We only have amd64 images for the three OS `Oracle`, `RHEL` and `Rockylinux`. Should we rollback to 9.3 instead? We can update them to 9.4 while releasing v1.7.1. WDYT?
> 
> cc @innobead @khushboo-rancher

Sounds good to me, as they are unavailable now.

The general guidance is to check if need to bump versions in each feature release. Let's make sure it.

**Pull request merged**